mod/CheckHintData.cs

```diff
@@ -0,0 +1,124 @@
+        public static readonly Color ProgressionColor = new(1.5f, 0.5f, 1.5f, 1);
+        public static readonly Color FoundColor = new(1.2f, 1.2f, 1.2f, 1);
+
+        public static Material ChildTextMat
+        {
+            get
+            {
+                if (childTextMat == null)
+                    childTextMat = Locator.GetAstroObject(AstroObject.Name.CaveTwin).transform
+                        .Find("Sector_CaveTwin/Sector_SouthHemisphere/Sector_SouthUnderground/Sector_City/Interactables_City/Arc_CT_City_KidDirectionToFossil_1/Arc 1")
+                        .GetComponent<Renderer>().material;
+                return childTextMat;
+            }
+        }
+
+        public static Material NormalTextMat
+        {
+            get
+            {
+                if (normalTextMat == null)
+                    normalTextMat = Locator.GetAstroObject(AstroObject.Name.TimberHearth).transform
+                        .Find("Sector_TH/Sector_Village/Sector_Observatory/Interactables_Observatory/NomaiEyeExhibit/NomaiEyePivot/Arc_TH_Museum_EyeSymbol/Arc 1")
+                        .GetComponent<Renderer>().material;
+                return normalTextMat;
+            }
+        }
+
+        private static Material childTextMat;
+        private static Material normalTextMat;
+
+        private Renderer rend;
+
+        public Color NomaiWallColor()
+        {
+            if (HasBeenFound) return FoundColor;
+            switch (Importance)
+            {
+                case CheckImportance.Junk:
+                    return JunkColor;
+                case CheckImportance.Useful:
+                    return UsefulColor;
+                case CheckImportance.Progression:
+                    return ProgressionColor;
+                default:
+                    {
+                        int rnd = Random.Range(0, 3);
+                        switch (rnd)
+                        {
+                            case 0:
+                                return JunkColor;
+                            case 1:
+                                return UsefulColor;
+                            default:
+                                return ProgressionColor;
+                        }    
+                    }
+            }
+        }
+
+        public void SetImportance(CheckImportance importance)
+        {
+            if ((int)Importance < (int)importance)
+            {
+                Importance = importance;
+            }
+        }
+
+        public void DetermineImportance(Location loc)
+        {
+            rend = GetComponent<Renderer>();
+
+            Locations.Add(loc);
+            if (APRandomizer.APSession.Locations.AllLocationsChecked.Contains(LocationNames.locationToArchipelagoId[loc])) HasBeenFound = true;
+
+            if (Importance != CheckImportance.Trap)
+            {
+                if (rend.material.name.Contains("TextChild")) IsChildText = true;
```

Comment:
IsChildText doesn't appear to be used outside of this component, or even this method. Can we change it to a local variable?

Reply:
sure done